pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -439,6 +439,8 @@ type SCMProviderGeneratorGitlab struct {
 	TokenRef *SecretRef `json:"tokenRef,omitempty" protobuf:"bytes,4,opt,name=tokenRef"`
 	// Scan all branches instead of just the default branch.
 	AllBranches bool `json:"allBranches,omitempty" protobuf:"varint,5,opt,name=allBranches"`
+	// Allow self-signed TLS / Certificates; default: false
+	Insecure bool `json:"insecure,omitempty" protobuf:"varint,6,opt,name=insecure"`
```

Comment:
To be super clear about what's insecure, can we call it `tlsInsecureSkipVerify`?

Reply:
I think we should keep it `Insecure` to match what we already have for Gitea today. WDYT?

Reply:
Good point, agreed.